### PR TITLE
Add ability to use glob patterns for source configs

### PR DESCRIPTION
**Pull request by r-ash**
This PR will
* Support sourcing files using pattern matching

### CODE REVIEW

File: R/dettl_config.R [hunk 1]
```diff
@@ -189,10 +189,40 @@ read_fields <- function(fields, config, env) {
...
+#'
+#' @keywords internal
+expand_wildcards <- function(sources, path) {
+  all_files <- c()
+  for (source in sources) {
+    files <- Sys.glob(file.path(path, source))
+    if (length(files) > 0) {
+      all_files <- c(all_files, files)
+    } else {
+      message(sprintf("No files found matching file pattern %s", source))
```
Comment (richfitz) [inline R/dettl_config.R]: I feel that this should be an error - if not then you could just use:
```r
expand_wildcards <- function(sources, path) {
  Sys.glob(file.path(sources, path))
}
```

File: R/dettl_config.R [hunk 2]
```diff
@@ -189,10 +189,40 @@ read_fields <- function(fields, config, env) {
...
+#'
+#' @param sources List of configured file paths some of which may contain
+#' wildcards. Relative to the root of the import.
+#' @param path Normalised path to the import directory.
+#'
+#' @return List of files to be used as sources
+#'
+#' @keywords internal
+expand_wildcards <- function(sources, path) {
+  all_files <- c()
```
Comment (richfitz) [inline R/dettl_config.R]: This pattern (growing a vector) is not very R-ish: more typical would be

```r
expand1 <- function(x) {
  files <- Sys.glob(x)
  if (length(files) == 0) {
    message(sprintf("No files found matching file pattern %s", x))
  }
  files
}

expand_wildcards <- function(sources, path) {
  unlist(lapply(sources, expand1, path), FALSE, FALSE)
}
```

File: R/dettl_config.R [hunk 3]
```diff
@@ -189,10 +189,40 @@ read_fields <- function(fields, config, env) {
...
+  for (source in sources) {
+    files <- Sys.glob(file.path(path, source))
```
Comment (richfitz) [inline R/dettl_config.R]: If you want to separate out the path part from the pattern matching, the other option is to use `withr::with_dir` to change working directory before processing the patterns - this usually results in cleaner code

**Review by richfitz — CHANGES_REQUESTED** [review]

**Comment by r-ash** [comment]
Thanks for the feedback above. I've addressed each of those points.

**Review by richfitz — APPROVED** [review]

**Pull request merged**